### PR TITLE
Compare datetimes instead of strings in _fixes/cmip5/test_access1_X.py

### LINKED ISSUE #1158: cf-units=2.1.5 for OSX still preserves older version behaviour (<2.1.4)
`cf-units=2.1.5` installed in the esmvaltool conda env on OSX as seen from the [GA test](https://github.com/ESMValGroup/ESMValCore/runs/2759882084?check_suite_focus=true) preserves the behavior of `num2date` from an older version, <2.1.4. We need to raise this with the `cf-units` develsm @bjlittle this is a SciTools package, would you be able to look into it maybe? :beer: 

Results for `cf_units.num2date(time.points, time.units.name, time.units.calendar)[0].strftime('%Y%m%d%H%M')` for Linux:
- Behavior for 2.1.4:  "  30001161200"
- Behavior for 2.1.5:  "30001161200"

I remember that "030001161200" result was when we were using 2.1.2 or something similar.

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Got rid of `num2date` calls that change the outputted string formatted date everytime there's a new release of `cf_units` and compare datime objects straight out the box.

Closes #1158 

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by zklaus** [comment]
As @trexfeathers pointed out in the issue, we don't know what we will get back from num2date. That maybe `datetime.datetime` or `cftime.datetime`. That makes the comparison with always `datetime.datetime` fragile, particularly since we use a date long before the Gregorian calendar. Maybe we should use the suggested `only_use_cftime_datetimes=True` kwarg to num2date and import cftime.datetime instead?

**Comment by valeriupredoi** [comment]
@zklaus fragile is good for testing - that prompts us to look into what's the problem. The thing is, we don't use only `cf_time.datetime` - we use mostly `datetime.datetime` in Core, so a test fail would give us a hint there is an issue with the comparison that might stem from different behaviour of `cf_time/units` vs the `datetime` module - I think it's good like this, what do you think?

**Comment by zklaus** [comment]
Alright. Sounds good. Let me do a formal review.

File: .github/workflows/action-test.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ on:
   push:
     branches:
     - main
+    - compare_datetimes
```
Comment (zklaus) [inline .github/workflows/action-test.yml:24]: Remove this?

**Review by zklaus — CHANGES_REQUESTED** [review]

**Comment by zklaus** [comment]
Could you please also go through the checklist and assign labels?

File: .github/workflows/action-test.yml (hunk 1 above)
Comment (valeriupredoi) [reply .github/workflows/action-test.yml:24]: on it right after the last OSX test passes

**Comment by valeriupredoi** [comment]
all tests had passed, removed the call to test on branch, ready to go go Klaus :checkered_flag:

**Review by zklaus — APPROVED** [review]
Just finish up the checklist please, possibly by removing or striking out not applicable items.

**Comment by valeriupredoi** [comment]
done, cheers for the review, Klaus! I think we're now good in terms of all the test fails, on various platforms, a release will be clean :1st_place_medal:

**Comment by zklaus** [comment]
Cheers V, great work these days! (And all others too, of course :stuck_out_tongue_closed_eyes: )

**Comment by valeriupredoi** [comment]
:beer: :beer: :beer:

**Pull request merged**